### PR TITLE
make sure that we get the correct user from the path

### LINKED ISSUE #13758: Moving files from a remote shared folder in an encrypted server to a not encrypted server using webdav fails.
### Steps to reproduce

Having two servers server1 and server2, server1 is encrypted and server2 isn't.

1. Share a folder with files from server1 to server2.
2. Using webdav interface of server2 try to move a file from remote shared folder located in server1 to a local folder of server2. 

### Expected behaviour
File is moved correctly.

### Actual behaviour
Move fails and it returns:
503 Service Unavailable

### Server configuration
**Operating system**:
Ubuntu 14.04 LTS

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
{"installed":true,"maintenance":false,"version":"8.0.0.4","versionstring":"8.0 beta 2","edition":""}

**Updated from an older ownCloud or fresh install:**
fresh

**List of activated apps:**
Encryption in server1.

**The content of config/config.php:**

```
Insert your config.php content here
(Without the database password and passwordsalt)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes in server1

### Client configuration
**Browser:**
Chrome

### PR DESCRIPTION
**Pull request by schiessle**
In case of a remote share no user will be logged in if a file gets deleted/moved. To be able to get the correct size we need to get the user from the path. Otherwise deleting and moving files in remote shared folders will not work.

fix #13758

### CODE REVIEW

**Comment by schiesbn** [comment]
Tested and verified that it works, here: https://github.com/owncloud/core/issues/13758#issuecomment-72205627 ... Count as one :+1: 

One more reviewer needed.

File: apps/files_encryption/lib/helper.php [hunk 1]
```diff
@@ -293,8 +293,8 @@ public static function getUser($path) {
 	public static function getUserFromPath($path) {
 		$split = self::splitPath($path);
 
-		if (count($split) > 3 && (
-			$split[2] === 'files' || $split[2] === 'files_versions' || $split[2] === 'cache')) {
+		if (count($split) > 2 && (
+			$split[2] === 'files' || $split[2] === 'files_versions' || $split[2] === 'cache' || $split[2] === 'files_trashbin')) {
```
Comment (LukasReschke) [inline apps/files_encryption/lib/helper.php]: :see_no_evil:

**Comment by schiesbn** [comment]
Would be great if we could get it in for 8.0 if it is not already to late... Just one +1 and clicking on the merge button is missing ;-)

cc @DeepDiver1975 @MorrisJobke @karlitschek

**Comment by LukasReschke** [comment]
Okay... But let me state that at some point we really need to get rid of such error-prone magic ;-)

:+1:

**Comment by schiesbn** [comment]
> Okay... But let me state that at some point we really need to get rid of such error-prone magic

With Encryption 2.0 we will have enough opportunities to clean the code up. For a bug fix one day before the release I really don't want to mess up the code. For a gut feeling I would say that we can skip the complete if-statement. On the other hand the statement might have been introduced for good reasons and I don't want to risk introducing new (more complex) bugs while fixing a quite simple bug.

**Pull request merged**